File: src/main/java/com/codecups/app/model/Product.java
package com.codecups.app.model;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

/**
 * Copyright CodeCups
 * Created by Niko on 14 April 2021
 */

@Entity (name = "products")
@Getter
@Setter
public class Product implements Serializable {

    @Getter(AccessLevel.NONE)
    @Setter(AccessLevel.NONE)
    private static final long serialVersionUID = -4157715894287389876L;

    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String productId;

    @Column(nullable = false, length = 50)
    private String name;

    private String description;

    @Column(nullable = false)
    private double price;

    private String image;

    @ManyToMany
    @JoinTable(name = "products_orders",
            joinColumns = {@JoinColumn(name = "order_id", referencedColumnName = "id")},
            inverseJoinColumns = {@JoinColumn(name = "product_id", referencedColumnName = "id")})
    private List<Order> orders;
}

File: src/main/java/com/codecups/app/web/codecups/src/app/models/product.ts
export class Product {
  id: string;
  name: string;
  description: string;
  price: number;
  imageUrl: string;


  constructor(id: string, name: string, description = '', price = 0, imageUrl = 'https://dummyimage.com/600x400/55595c/fff') {
    this.id = id;
    this.name = name;
    this.description = description;
    this.price = price;
    this.imageUrl = imageUrl;
  }
}
